napari/_app_model/actions/_toggle_action.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from app_model.types import Action, ToggleRule
+
+if TYPE_CHECKING:
+    from ...viewer import Viewer
+
+
+class ViewerToggleAction(Action):
+    """Action subclass that toggles a boolean viewer (sub)attribute on trigger.
+    Parameters
+    ----------
+    id : CommandId
+        The id of the action.
+    title : str
+        The title of the action.
+    viewer_attribute : str
+        The attribute of the viewer to toggle. (e.g. 'axes')
+    sub_attribute : str
+        The attribute of the viewer attribute to toggle. (e.g. 'visible')
+    **kwargs
+        Additional keyword arguments to pass to the Action constructor.
+    Examples
+    --------
+    >>> action = ViewerToggleAction(
+    ...     id='some.command.id',
+    ...     title='Toggle Axis Visibility',
+    ...     viewer_attribute='axes',
+    ...     sub_attribute='visible',
+    ... )
+    """
+
+    def __init__(
+        self,
+        *,
+        id: str,
+        title: str,
+        viewer_attribute: str,
+        sub_attribute: str,
+        **kwargs,
+    ):
+        def get_current(viewer: Viewer):
+            """return the current value of the viewer attribute"""
+            attr = getattr(viewer, viewer_attribute)
+            return getattr(attr, sub_attribute)
+
+        def toggle(viewer: Viewer):
+            """toggle the viewer attribute"""
+            attr = getattr(viewer, viewer_attribute)
+            setattr(attr, sub_attribute, not getattr(attr, sub_attribute))
```

Comment:
why all these functions are declared here, not are method of a declared class?

Reply:
Sorry this does not need to be here. I should have made it clear that this is based off of #4826 and some things have been copied over as they were needed for this PR to work.  

> The file napari/_app_model/actions/_toggle_action.py is not even imported during tests.

I initially thought `ViewerToggleAction` was needed but I later realised that the actions of the items in the `Window` menu need qt so `ViewerToggleAction` is not needed. Please let me know if I have mis-understood this.

I will remove this file for now.